disnake/ext/commands/base_core.py

```diff
@@ -190,12 +198,51 @@ def __init__(self, func: CommandCallback, *, name: str = None, **kwargs):
         self._max_concurrency: Optional[MaxConcurrency] = max_concurrency
 
         self.cog: Optional[Cog] = None
-        self.guild_ids: Optional[List[int]] = None
+        self.guild_ids: Optional[Tuple[int, ...]] = None
         self.auto_sync: bool = True
 
         self._before_invoke: Optional[Hook] = None
         self._after_invoke: Optional[Hook] = None
 
+    def _ensure_assignment_on_copy(self, other: AppCommandT) -> AppCommandT:
+        other._before_invoke = self._before_invoke
+        other._after_invoke = self._after_invoke
+        if self.checks != other.checks:
+            other.checks = self.checks.copy()
+        if self._buckets.valid and not other._buckets.valid:
+            other._buckets = self._buckets.copy()
+        if self._max_concurrency != other._max_concurrency:
+            # _max_concurrency won't be None at this point
+            other._max_concurrency = cast(MaxConcurrency, self._max_concurrency).copy()
+
+        try:
+            other.on_error = self.on_error
+        except AttributeError:
+            pass
+
+        return other
+
+    def copy(self: AppCommandT) -> AppCommandT:
+        """Create a copy of this application command.
+
+        Returns
+        -------
+        :class:`InvokableApplicationCommand`
+            A new instance of this application command.
+        """
+        copy = type(self)(self.callback, **self.__original_kwargs__)
+        return self._ensure_assignment_on_copy(copy)
+
+    def _update_copy(self: AppCommandT, kwargs: Dict[str, Any]) -> AppCommandT:
+        if kwargs:
+            kw = kwargs.copy()
+            kw.update(self.__original_kwargs__)
+            print(kw)
```

Comment:
this shouldn't be here

Reply:
https://github.com/DisnakeDev/disnake/pull/501/commits/0f56578ab610f7deba8c937d1be0ba3f88bf4dc8